packages/backend/server/src/plugins/copilot/providers/openai.ts

```diff
@@ -0,0 +1,194 @@
+import assert from 'node:assert';
+
+import { ClientOptions, OpenAI } from 'openai';
+
+import {
+  ChatMessage,
+  ChatMessageRole,
+  CopilotCapability,
+  CopilotProviderType,
+  CopilotTextToEmbeddingProvider,
+  CopilotTextToTextProvider,
+} from '../types';
+
+export class OpenAIProvider
+  extends OpenAI
```

Comment:
The provider is limited by interfaces, nonsense to extends OpenAI, make it a field instead